inc/dbutils.class.php

```diff
@@ -1709,52 +1707,75 @@ public function autoName($objectName, $field, $isTemplate, $itemtype, $entities_
                   'Printer'
                ];
 
+               $subqueries = [];
                foreach ($types as $t) {
                   $table = $this->getTableForItemType($t);
-                  $query .= ($first ? "SELECT " : " UNION SELECT  ")." $field AS code
-                           FROM `$table`
-                           WHERE `$field` LIKE '$like'
-                                 AND `is_deleted` = 0
-                                 AND `is_template` = 0";
+                  $criteria = [
+                     'SELECT' => ["$field AS code"],
+                     'FROM'   => $table,
+                     'WHERE'  => [
+                        $field         => ['LIKE', $like],
+                        'is_deleted'   => 0,
+                        'is_template'  => 0
+                     ]
+                  ];
 
                   if ($CFG_GLPI["use_autoname_by_entity"]
                      && ($entities_id >= 0)) {
-                     $query .=" AND `entities_id` = '$entities_id' ";
+                     $criteria['WHERE']['entities_id'] = $entities_id;
                   }
 
-                  $first = 0;
+                  $subqueries[] = new \QuerySubQuery($criteria);
                }
 
-               $query = "SELECT CAST(SUBSTRING(code, $pos, $len) AS unsigned) AS no
-                        FROM ($query) AS codes";
-
+               $criteria = [
+                  'SELECT' => [
+                     new \QueryExpression(
+                        "CAST(SUBSTRING(".$DB->quoteName('code').", $pos, $len) AS " .
+                        $DB->quoteName('unsigned') . ") AS " . $DB->quoteName('no')
```

Comment:
unsigned is not a field name, but a variable type
```suggestion
                        . "unsigned) AS " . $DB->quoteName('no')
```